### PR TITLE
add delete method for dialogs

### PR DESCRIPTION
**Pull request by nailerNAS**

### CODE REVIEW

File: telethon/tl/custom/dialog.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 from . import Draft
 from .. import TLObject, types
+from ..functions.channels import LeaveChannelRequest
+from ..functions.messages import DeleteHistoryRequest
```
Comment (Lonami) [inline telethon/tl/custom/dialog.py]: Change this by editing the line above to be `from .. import TLObject, types, functions`.

File: telethon/tl/custom/dialog.py [hunk 2]
```diff
@@ -83,8 +87,8 @@ def __init__(self, client, dialog, entities, messages):
 
         self.is_user = isinstance(self.entity, types.User)
         self.is_group = (
-            isinstance(self.entity, types.Chat) or
-            (isinstance(self.entity, types.Channel) and self.entity.megagroup)
+                isinstance(self.entity, types.Chat) or
```
Comment (Lonami) [inline telethon/tl/custom/dialog.py]: Why the extra indent?

File: telethon/tl/custom/dialog.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
 from . import Draft
 from .. import TLObject, types
+from ..functions.channels import LeaveChannelRequest
+from ..functions.messages import DeleteHistoryRequest
 from ... import utils
+from ...errors import PeerIdInvalidError
```
Comment (Lonami) [inline telethon/tl/custom/dialog.py]: Can you also do `from ... import utils, errors`?

File: telethon/tl/custom/dialog.py [hunk 4]
```diff
@@ -96,6 +100,13 @@ def __init__(self, client, dialog, entities, messages):
         return await self._client.send_message(
             self.input_entity, *args, **kwargs)
 
+    async def delete(self):
+        try:
+            await self._client(DeleteHistoryRequest(self, 0))
+
```
Comment (Lonami) [inline telethon/tl/custom/dialog.py]: Unnecessary empty line.

File: telethon/tl/custom/dialog.py [hunk 5]
```diff
@@ -96,6 +100,13 @@ def __init__(self, client, dialog, entities, messages):
...
+        except PeerIdInvalidError:
```
Comment (Lonami) [inline telethon/tl/custom/dialog.py]: You have the `is_channel` property… why don't you use it instead making an invalid API call?

File: telethon/tl/custom/dialog.py [hunk 6]
```diff
@@ -96,6 +100,13 @@ def __init__(self, client, dialog, entities, messages):
...
+            await self._client(LeaveChannelRequest(self))
```
Comment (Lonami) [inline telethon/tl/custom/dialog.py]: Can you pass `self.input_entity` instead? It's more efficient.

File: telethon/tl/custom/dialog.py [hunk 7]
```diff
@@ -96,6 +100,13 @@ def __init__(self, client, dialog, entities, messages):
         return await self._client.send_message(
             self.input_entity, *args, **kwargs)
 
+    async def delete(self):
+        try:
+            await self._client(DeleteHistoryRequest(self, 0))
```
Comment (Lonami) [inline telethon/tl/custom/dialog.py]: Can you pass `self.input_entity` instead? It's more efficient.

File: telethon/tl/custom/dialog.py [hunk 8]
```diff
@@ -96,6 +96,12 @@ def __init__(self, client, dialog, entities, messages):
         return await self._client.send_message(
             self.input_entity, *args, **kwargs)
 
+    async def delete(self):
+        if self.is_channel:
+            await self._client(functions.channels.LeaveChannelRequest(self.input_entity))
+        else:
```
Comment (Lonami) [inline telethon/tl/custom/dialog.py]: I guess we need an `elif self.is_group: functions.messages.DeleteChatUserRequest(self.entity.id, types.InputPeerSelf())` here.

Comment (Kyle2142) [reply telethon/tl/custom/dialog.py]: To remove the exited group from dialogs : `functions.messages.DeleteHistoryRequest(self.entity.id, 0, just_clear=False)`

**Pull request merged**